librealsense: init at 2.11.0

brian-dawn opened the pull request:
###### Motivation for this change

The librealsense library is used to drive a series of depth cameras from Intel. The package doesn't depend on much and would be helpful to those of us who use nix for C++ building.

This is my first contribution to nixpkg so please let me know if I did anything wrong! Unfortunately testing the library is difficult as it requires having a realsense camera to really test it. I was able to test it on my machine using Ubuntu 16.04 and an intel D435.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

brian-dawn committed:
librealsense: init at 2.11.0

jtojnar commented on pkgs/development/libraries/librealsense/default.nix:
```diff
@@ -0,0 +1,31 @@
+  enableParallelBuilding = true;
+
+  preConfigure = ''
+    cmakeFlags="$cmakeFlags -DBUILD_EXAMPLES=false"
```
You can use `cmakeFlags = [ "-DBUILD_EXAMPLES=false" ]` attribute in nix.

jtojnar commented on pkgs/development/libraries/librealsense/default.nix:
```diff
@@ -0,0 +1,31 @@
+    cmake
+  ];
+
+  enableParallelBuilding = true;
```
cmake builds in parallel by default.

jtojnar commented on pkgs/development/libraries/librealsense/default.nix:
```diff
@@ -0,0 +1,31 @@
+  };
+
+  nativeBuildInputs = [
+    libusb
```
`nativeBuildInputs` targets the `build` platform during [cross-compilation](https://nixos.org/nixpkgs/manual/#chap-cross). Unless you use some utility from `libusb` during, it should  be moved to `buildInputs`.

jtojnar commented on pkgs/development/libraries/librealsense/default.nix:
```diff
@@ -0,0 +1,31 @@
+  nativeBuildInputs = [
+    libusb
+    pkgconfig
+    cmake
```
When using `cmake`, it is a good idea to add `ninja` as well – it will run slightly faster than the Make backend.

brian-dawn commented:
Thanks for the review @jtojnar I think I should have addressed all your comments.

It seems I can't access your review anymore, the contributors guide recommended avoiding multiple commits so I have squashed it down. I'm not used to GitHub so my apologies if that was the wrong workflow.
